.github/workflows/tests-main.yml

```diff
@@ -30,7 +30,7 @@ jobs:
         python -m pip install --upgrade pip
         # install PEFT & transformers from source
         pip install -U git+https://github.com/huggingface/peft.git
-        pip install -U git+https://github.com/huggingface/transformers.git
+        pip install -U git+https://github.com/huggingface/transformers.git@f339042b0b8bdc0b57a70d37f67cafbea960a2ab
```

Comment:
Not for this PR. But do you think we should add a CI workflow that tests against our lowest supported version of transformers? This may have picked up the issue with the examples being broken on main when `processing_class` was added?

Reply:
Good point. For ref, #2298 aims to make tests clearer/cleaner. I'll extend the work with another PR that implements your suggestion.

Reply:
pip doesn't have an option to prefer the min version yet https://github.com/pypa/pip/issues/8085
Let's do it manually

Reply:
> There is another recent breaking change. I set this commit to make the CI run anyway. I will fix the new breaking change in a followup PR

This other breaking change is solved here: #2302

Reply:
Done in #2303